### PR TITLE
added Kong ingress example

### PR DESCRIPTION
**Pull request by gAmUssA**
Kind documentation included examples from a popular Kubernetes ingress controller.
This PR adds a doc/tutorial of how to enable Kong Ingress Controller with `kind`.

### CODE REVIEW

File: site/content/docs/user/ingress.md [hunk 1]
```diff
@@ -112,6 +113,51 @@ Refer to [Using Ingress](#using-ingress) for a basic example usage.
 
 Additional information about Contour can be found at: [projectcontour.io](https://projectcontour.io)
 
+### Ingress Kong
+
+Deploy [Kong Ingress Controller (KIC)](https://docs.konghq.com/kubernetes-ingress-controller/2.1.x/concepts/design/).
+
+{{< codeFromInline lang="bash" >}}
+kubectl apply -f bit.ly/k4k8s
```
Comment (aojea) [inline site/content/docs/user/ingress.md:121]: bit.ly ? Don't we have a GitHub or corporate URL ?

Comment (gAmUssA) [reply site/content/docs/user/ingress.md:121]: @aojea it's actually GH link https://raw.githubusercontent.com/Kong/kubernetes-ingress-controller/master/deploy/single/all-in-one-dbless.yaml.
Let me know if you want me to include a full link.

Comment (aojea) [reply site/content/docs/user/ingress.md:121]: If you don't mind? I think is most transparent

Comment (gAmUssA) [reply site/content/docs/user/ingress.md]: sure thing! pushed an update in 17e8fa2

**Comment by aojea** [comment]
better squash all commits, please

**Comment by gAmUssA** [comment]
> better squash all commits, please
✅ @aojea

File: site/content/docs/user/ingress.md [hunk 2]
```diff
@@ -112,6 +113,51 @@ Refer to [Using Ingress](#using-ingress) for a basic example usage.
...
+Apply kind specific patch to change service type to `NodePort`:
+
+```json
+{{% readFile "static/examples/ingress/kong/service.patch.json" %}}
+```
+
+Apply it by running:
+
+{{< codeFromInline lang="bash" >}}
+kubectl patch service -n kong ingress-kong -p '{{< minify file="static/examples/ingress/contour/service.patch.json" >}}'
```
Comment (aojea) [inline site/content/docs/user/ingress.md:145]:
```suggestion
kubectl patch service -n kong ingress-kong -p '{{< minify file="static/examples/ingress/kong/service.patch.json" >}}'
```

File: site/static/examples/ingress/kong/deployment.patch.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            ]
+          }
+        ],
+        "nodeSelector": {
+          "ingress-ready": "true"
+        },
+        "tolerations": [
+          {
+            "effect": "NoSchedule",
+            "key": "node-role.kubernetes.io/master",
```
Comment (aojea) [inline site/static/examples/ingress/kong/deployment.patch.json:30]: @neolit123 advice here please

Comment (neolit123) [reply site/static/examples/ingress/kong/deployment.patch.json:30]: this should be either:
https://github.com/kubernetes-sigs/kind/blob/main/site/static/examples/ingress/contour/patch.json#L8-L18

or:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/phases/addons/proxy/manifests.go#L111-L112

the former tolerates only the CP node taints for kubeadm 1.24, the latter tolerate all taints (i.e. a wildcard).

File: site/static/examples/ingress/kong/deployment.patch.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+            "operator": "Equal"
+          }
+        ]
+      }
+    }
+  }
+}
```
Comment (neolit123) [inline site/static/examples/ingress/kong/deployment.patch.json:37]: these json files have missing new lines.

**Comment by gAmUssA** [comment]
Updated accordingly, squashed all commits. @neolit123 @aojea

**Comment by aojea** [comment]
> Updated accordingly, squashed all commits. @neolit123 @aojea

https://github.com/kubernetes-sigs/kind/pull/2603#discussion_r793651829

File: site/static/examples/ingress/kong/deployment.patch.json (hunk 4 above)
Comment (gAmUssA) [reply site/static/examples/ingress/kong/deployment.patch.json]: @neolit123 not quite sure where new lines are missing?

**Comment by aojea** [comment]
/lgtm
/approve

I don't have time to test it and I don't think any of the maintainers have time to do it, so we expect the other projects linked from KIND to maintain their scripts and manifests.

Thanks

**Comment by gAmUssA** [comment]
@aojea Thank you!
Feed free to tag me anytime if users will report problem with Kong Ingress. 
Cheers,
Vik

**Pull request merged**